sccm-technicalpreview/azuredeploy.json

```diff
@@ -228,17 +213,17 @@
     },
     {
       "type": "Microsoft.Network/networkInterfaces",
-      "name": "[concat(toLower(parameters('prefix')),toLower(variables('vmname')[copyIndex()]), '-ni')]",
-      "apiVersion": "2018-08-01",
+      "name": "[concat(toLower(parameters('prefix')),toLower(if(less(copyIndex(),length(variables('vmInfo'))),variables('vmInfo')[variables('vmRole')[copyIndex()]].name,variables('siteInfo')[variables('siteRole')[sub(copyIndex(),length(variables('vmInfo')))]].name)), '-ni')]",
```

Comment:
this is a really hard pattern to reason over - let's break these into multiple loops (one for vmInfo and one for siteInfo) to improve the readability.  (will need to do this for the other resources as well)